### PR TITLE
Add mosaic script

### PR DESCRIPTION
**Pull request by alexgleith**
Adds a new command line tool that we can use to create low-res mosaics easily.

### CODE REVIEW

File: deafrica/data/create_mosaic.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import click
+import pystac
+from datacube import Datacube
+from deafrica.utils import setup_logging
+from odc.algo import save_cog
+from odc.aws import s3_dump
+from rio_stac import create_stac_item
+
+
+def create_asset(href):
```
Comment (NikitaGandhi) [inline deafrica/data/create_mosaic.py]: can this function be inline if it has only one line in a function?

Comment (alexgleith) [reply deafrica/data/create_mosaic.py]: Yep

File: deafrica/data/create_mosaic.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+@click.command("create-mosaic")
+@click.option("--product", type=str, default="gm_ls8_annual")
+@click.option("--out-product", type=str, default=None)
+@click.option("--year", type=str, default="2020")
+@click.option("--bands", type=str, default="red")
+@click.option("--resolution", type=int, default=120)
+@click.option(
+    "--s3-output-file",
+    type=str,
+    default="s3://deafrica-data-dev-af/alex-test-mosaics/test_120m_red.tif",
```
Comment (NikitaGandhi) [inline deafrica/data/create_mosaic.py]: default is set to alex path in bucket?

Comment (alexgleith) [reply deafrica/data/create_mosaic.py]: Changed to a more obvious example :-)

**Review by NikitaGandhi — APPROVED** [review]

**Pull request merged**